encrypted-media-respec.html

```diff
@@ -605,8 +605,112 @@
 
           <section>
             <h5>Get Supported Configuration</h5>
-            <p>Given a <a def-id="keysystems"></a> implementation <var>implementation</var>, <a>MediaKeySystemConfiguration</a> <var>candidate configuration</var>, and <var>origin</var>, this algorithm returns a supported configuration or <code>null</code> as appropriate.</p>
-            <p class="note">Unrecognized dictionary members in <var>candidate configuration</var> are ignored per [[WebIDL]] and will never reach this algorithm. Thus, they cannot be considered as part of the configuration.</p>
+            <p>Given a <a def-id="keysystems"></a> implementation <var>implementation</var>, <a>MediaKeySystemConfiguration</a> <var>candidate configuration</var>, and <var>origin</var>, this algorithm returns a supported configuration or <code>NotSupported</code> as appropriate.</p>
+            <p class="note">Unrecognized dictionary members in <var>candidate configuration</var> are ignored per [[WebIDL]] and will never reach this algorithm. Thus, they cannot be considered as part of the configuration.
+            </p>
+            <div class="note">
+              <p>
+              For certain configurations, it may be required to obtain user consent or inform the user. User Agents have some flexibility to determine
+              whether consent is required for a specific configuration and whether such consent may also apply to other configurations. For example,
+              consent to one configuration may also imply consent for less powerful, more restricted configurations. Equally, a denial of consent for
+              one configuration may imply denial of consent for more powerful, less restricted configurations.
+              </p>
+              <p>
+              Supported configurations, including supported audio and video codecs, may depend on availability of optional capabilities such as
+              <a def-id="distinctive-identifier"></a>s and persistent state. The following algorithm iteratively tries configurations with different
+              requirement values for these optional configuration elements until consent is achieved.
+              </p>
+              <p>
+              User Agents should persist consent responses at least for the duration of the <a def-id="requestMediaKeySystemAccess"></a>
+              algorithm in order to avoid repeated requests to the user for similar configurations.
+              </p>
+            </div>
+            <ol>
+              <li>
+                <p>Let <var>requirements</var> be the list [ <code>distinctiveIdentifier</code>, <code>persistentState</code> ].</p>
```

Comment:
> The "worst case" for the algorithm is that both are "optional". In that case, an implementation might choose [something that results in] "required" for DI and prompt. (PS could be either value.) If the consent is declined, that almost certainly means that DI should not be "not-allowed".

Do you mean *should* be "not-allowed" ? (Consent was denied).

> Perhaps we can simplify this by allowing implementations to request consent only once per configuration. (As you said, how this is handled is up to the implementation.)

Certainly we can simplify by *requiring* implementations to request consent only once. The only problem is that the consent is to the use (or not) of DI and PS *in association with _accumulated configuration_*. If all the subsequent configurations tried are subsets of _accumulated configuration_ (from a consent perspective) we are ok: the original consent still applies. But if not, we are stuck.

For example, the site marks both DI and PS as optional. The implementation chooses codecs and other options. that require both. The user consents only to DI "in association with those codecs and other options". So, we try again with DI required and PS not allowed. The set of codecs and other options now chosen is different. Does the prior consent still apply ? Perhaps there is a robustness level that implies hardware DRM that works without PS but that if you want to use a lower level (sw) that requires PS and perhaps the UA distinguishes HW and SW from a consent perspective ?